File: bm_db_android_java/app/src/main/java/steffen/de/bm_db_android_java/Database/DataDao.java
package steffen.de.bm_db_android_java.Database;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

import steffen.de.bm_db_android_java.Model.Data;

@Dao
public interface DataDao {

    @Query("SELECT * FROM data")
    List<Data> getAll();

    @Insert
    void insert(List<Data> dataList);

    @Delete
    void delete(Data data);
}

File: bm_db_android_kotlin/app/src/main/java/steffen/de/bm_db_android_kotlin/Database/DataDao.kt
package steffen.de.bm_db_android_kotlin.Database

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Delete
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query
import steffen.de.bm_db_android_kotlin.Model.Data

@Dao
interface DataDao {

    @get:Query("SELECT * FROM data")
    val all: MutableList<Data>

    @Insert
    fun insert(dataList: MutableList<Data>)

    @Delete
    fun delete(dataList: MutableList<Data>)
}